components/script/dom/window.rs

```diff
@@ -1356,7 +1402,9 @@ impl Window {
     }
 
     pub fn get_url(&self) -> Url {
-        (*self.Document().url()).clone()
+        let doc = self.Document();
+        let val = (*doc.url()).clone();
+        val
```

Comment:
I've opened up rust-lang/rust#37785 for this. In the meantime, add a comment linking to the relevant rust-lang issue and mention that this is a temporary fix to appease borrowck.